debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/Wal2JsonReplicationMessage.java

```diff
@@ -139,105 +149,208 @@ public String getName() {
      * @return the value; may be null
      */
     public Object getValue(String columnName, String columnType, Value rawValue, final PgConnectionSupplier connection) {
-        switch (columnType) {
+        // we support wal2json both before & after commit 0255f2ac, which means we may get old-style type names 
+        // (eg. int2, timetz, varchar, numeric, _int4)
+        // or new-style/verbose ones
+        // (eg. smallint; time with time zone; character varying (255); decimal (10,3), integer[])
+        
+        Matcher m = TYPE_PATTERN.matcher(columnType);
+        if (!m.matches()) {
+            LOGGER.error("Failed to parse columnType for {} '{}'", columnName, columnType);
+            throw new ConnectException(String.format("Failed to parse columnType '%s' for column %s", columnType, columnName));
+        }
+        String fullType = m.group("full");
+        String baseType = m.group("base").trim();
+        if (!Objects.toString(m.group("suffix"), "").isEmpty()) {
+            baseType = String.join(" ", baseType, m.group("suffix").trim());
+        }
+        String[] typeModifiers = Objects.toString(m.group("mod"), "").split(",\\s*", 0);  // TODO: use for decimal/etc precision
+        Boolean isArray = (m.group("array") != null);
+        
+        if (baseType.startsWith("_")) {
+            // old-style type specifiers use an _ prefix for arrays
+            // e.g. int4[] would be "_int4"
+            baseType = baseType.substring(1);
+            fullType = fullType.substring(1);
+            isArray = true;
+        }
+        
+        if (rawValue.isNull()) {
+            // nulls are null
+            return null;
```

Comment:
this path isn't covered by the tests. How to add?

Reply:
Could you do it like int his method - https://github.com/debezium/debezium/blob/master/debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java#L71 just user an `INSERT` statement that inserts nulls?